tests/unit/test_workflow_files.py

```diff
@@ -1832,87 +1835,45 @@ def test_detect_both_flow_and_suite(tmp_path):
-            id="No flow.cylc, suite.rc exists"
-        ),
-        pytest.param(
-            None, False, WorkflowFilesError, None,
-            id="No flow.cylc, no suite.rc"
-        ),
-    ]
-)
-@pytest.mark.parametrize(
-    'symlink_suiterc_arg',
-    [pytest.param(False, id="symlink_suiterc=False "),
-     pytest.param(True, id="symlink_suiterc=True ")]
-)
-def test_check_flow_file_symlink(
-    flow_file_target: Optional[str],
-    suiterc_exists: bool,
-    err: Optional[Type[Exception]],
-    expected_file: Optional[str],
-    symlink_suiterc_arg: bool,
-    tmp_path: Path, caplog: pytest.LogCaptureFixture
-) -> None:
-    """Test check_flow_file() when flow.cylc is a symlink or doesn't exist.
-
-    Params:
-        flow_file_target: Relative path of the flow.cylc symlink's target, or
-            None if the symlink doesn't exist.
-        suiterc_exists: Whether there is a suite.rc file in the dir.
-        err: Type of exception if expected to get raised.
-        expected_file: Which file's path should get returned, when
-            symlink_suiterc_arg is FALSE (otherwise it will always be
-            flow.cylc, assuming no exception occurred).
-        symlink_suiterc_arg: Value of the symlink_suiterc arg passed to
-            check_flow_file().
-    """
+def test_detect_both_flow_and_suite_symlinked(tmp_path):
+    """Test flow.cylc symlinked to suite.rc together in dir is permitted."""
+    (tmp_path / WorkflowFiles.SUITE_RC).touch()
     flow_file = tmp_path.joinpath(WorkflowFiles.FLOW_FILE)
-    suiterc = tmp_path.joinpath(WorkflowFiles.SUITE_RC)
-    tmp_path.joinpath('other-path').touch()
-    if suiterc_exists:
-        suiterc.touch()
-    if flow_file_target:
-        flow_file.symlink_to(flow_file_target)
+    flow_file.symlink_to(WorkflowFiles.SUITE_RC)
+    detect_both_flow_and_suite(tmp_path)
+
+
+def test_flow_symlinked_elsewhere_and_suite_present(tmp_path: Path):
+    """flow.cylc symlinked to suite.rc elsewhere, and suite.rc in dir raises"""
+    tmp_path.joinpath('some_other_dir').mkdir(exist_ok=True)
+    suite_file = tmp_path.joinpath('some_other_dir', WorkflowFiles.SUITE_RC)
+    suite_file.touch()
+    run_dir = tmp_path.joinpath('run_dir')
+    run_dir.mkdir(exist_ok=True)
+    flow_file = (run_dir / WorkflowFiles.FLOW_FILE)
+    flow_file.symlink_to(suite_file)
+    forbidden_external = is_forbidden(flow_file)
+    assert forbidden_external is False
+    (run_dir / WorkflowFiles.SUITE_RC).touch()
+    forbidden = is_forbidden(flow_file)
+    assert forbidden is True
+    with pytest.raises(WorkflowFilesError) as exc:
+        detect_both_flow_and_suite(run_dir)
+        assert str(exc.value) == (
+            "Both flow.cylc and suite.rc files are present in the "
+            "source directory. Please remove one and try again. "
+            "For more information visit: "
+            "https://cylc.github.io/cylc-doc/latest/html/7-to-8/summary.html"
+            "#backward-compatibility"
+        )
```

Comment:
As above (re: indentation) + simplify
```suggestion
    assert str(exc.value).startswith(
        "Both flow.cylc and suite.rc files are present in the "
        "source directory. Please remove one and try again. 
    )
```